### PR TITLE
Incremental seq2seq loss with batch averaging

### PR DESCRIPTION
**Pull request by jungokasai**
This change makes the seq2seq loss computation incremental over time steps and saves memory however long the target sentence is. Averaging of the loss is done accordingly on the fly so that we get the same loss as the original version. For the averaging purpose, the `average="sum"` option is added to `sequence_cross_entropy_with_logits`. There might be a cleaner way to handle this.  Also, this eliminates the softmax normalization for argmax prediction to avoid applying normalization twice (loss and prediction).

### CODE REVIEW

**Comment by matt-gardner** [comment]
@brendan-ai2, is it worth finishing this ourselves?  @jungokasai, we'll probably close this soon due to inactivity if we don't hear from you.

**Comment by brendan-ai2** [comment]
Seems like a nice change, but I'm a bit slammed right now. Would be best if @jungokasai can drive it forward.
